presto-kinesis/src/main/java/io/prestosql/plugin/kinesis/s3config/S3TableConfigClient.java

```diff
@@ -0,0 +1,243 @@
+        this.clientManager = requireNonNull(clientManager, "client manager object is null");
+        this.streamDescriptionCodec = requireNonNull(jsonCodec, "JSON codec object is null");
+
+        // If using S3 start thread that periodically looks for updates
+        if (kinesisConfig.getTableDescriptionLocation().startsWith("s3://")) {
+            this.bucketUrl = Optional.of(kinesisConfig.getTableDescriptionLocation());
+        }
+        else {
+            this.bucketUrl = Optional.empty();
+        }
+    }
+
+    /**
+     * Indicates this class is being used and actively reading table definitions from S3.
+     */
+    public boolean isUsingS3()
+    {
+        return bucketUrl.isPresent() && (bucketUrl.get().startsWith("s3://") || bucketUrl.get().equals("testloc"));
+    }
+
+    /**
+     * Main entry point to get table definitions from S3 using bucket and object directory
+     * given in the configuration.
+     * <p>
+     * For safety, an immutable copy built from the internal map is returned.
+     *
+     * @return
+     */
+    public Map<SchemaTableName, KinesisStreamDescription> getTablesFromS3()
+    {
+        updateTablesFromS3();
+        Collection<KinesisStreamDescription> streamValues = this.internalMap.values();
+        ImmutableMap.Builder<SchemaTableName, KinesisStreamDescription> builder = ImmutableMap.builder();
+        for (KinesisStreamDescription stream : streamValues) {
+            builder.put(new SchemaTableName(stream.getSchemaName(), stream.getTableName()), stream);
+        }
+        return builder.build();
+    }
+
+    /**
+     * Shutdown any periodic update jobs.
+     */
+    @Override
+    public void run()
+    {
+        if (isUsingS3() && updateTaskHandle != null) {
+            updateTaskHandle.cancel(true);
+        }
+        return;
+    }
+
+    @PostConstruct
+    protected void startS3Updates()
+    {
+        if (this.bucketUrl.isPresent()) {
+            ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
+            this.updateTaskHandle =
+                    scheduler.scheduleAtFixedRate(() -> updateTablesFromS3(), 5, 600, TimeUnit.SECONDS);
+        }
+    }
+
+    /**
+     * Call S3 to get the most recent object list.
+     * <p>
+     * This is an object list request to AWS in the given "directory".
+     *
+     * @return
+     */
+    protected List<S3ObjectSummary> getObjectSummaries()
+    {
+        AmazonS3Client s3client = this.clientManager.getS3Client();
+        AmazonS3URI directoryURI = new AmazonS3URI(this.bucketUrl.get());
+
+        List<S3ObjectSummary> returnList = new ArrayList<>();
+        try {
+            log.info("Getting the listing of objects in the S3 table config directory: bucket %s prefix %s :", directoryURI.getBucket(), directoryURI.getKey());
+            ListObjectsRequest req = new ListObjectsRequest().withBucketName(directoryURI.getBucket())
```

Comment:
Place `.withBucketName` on the next line.

Reply:
Also, don't abbreviate variable names.

Reply:
done